sdk/core/azure-core/src/main/java/com/azure/core/http/policy/DefaultRedirectStrategy.java

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A default implementation of {@link RedirectStrategy} that uses the provided try count, header name, http methods
+ * and status code to determine if request should be redirected.
+ */
+public final class DefaultRedirectStrategy implements RedirectStrategy {
+    private final ClientLogger logger = new ClientLogger(DefaultRedirectStrategy.class);
+
+    // Based on Stamp specific redirects design doc
+    private static final int DEFAULT_MAX_REDIRECT_ATTEMPTS = 10;
+    private static final String DEFAULT_REDIRECT_LOCATION_HEADER_NAME = "Location";
+    private static final int PERMANENT_REDIRECT_STATUS_CODE = 308;
+    private static final HashMap<Integer, HttpMethod> DEFAULT_ALLOWED_METHODS = new HashMap<Integer, HttpMethod>() {
+        {
+            put(HttpMethod.GET.ordinal(), HttpMethod.GET);
+            put(HttpMethod.HEAD.ordinal(), HttpMethod.HEAD);
+        }
+    };
+
+    private final int maxAttempts;
+    private final String locationHeader;
+    private final Map<Integer, HttpMethod> redirectMethods;
+    private final Set<String> attemptedRedirectUrls = new HashSet<>();
```

Comment:
This shouldn't be at the policy level. It should be per individual request. For e.g. request 1 goes to UrlA-> UrlB -> UrlC. Now, this set will contain UrlA, UrlB and UrlC. When another request is made from the client it would still start with UrlA because that's what the client is configured to use and this will abort the request without making the second request assuming that this URL was tried previously.

Reply:
Updated to be set on per request.